webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,312 @@
+            isLoaded: false,
+            ticketId: ticketID,
+        };
+    }
+
+    getIssueKey = () => {
+        let ticketID = '';
+        if (this.props.href.includes('selectedIssue')) {
+            ticketID = this.props.href.split('selectedIssue=')[1].split('&')[0];
+        }
+
+        if (!ticketID && this.props.href.includes('atlassian.net/browse')) {
+            ticketID = this.props.href.split('|')[0].split('?')[0].split('/browse/')[1];
+        }
+
+        return ticketID;
+    }
+
+    init() {
+        let instanceID = this.props.defaultUserInstanceID || '';
+
+        if (this.props.connectedInstances.length === 1) {
+            instanceID = this.props.connectedInstances[0].instance_id;
+        }
+
+        const ticketID = this.getIssueKey();
+        if (ticketID) {
+            this.props.getIssueByKey(ticketID, instanceID);
+        }
+    }
+
+    componentDidMount() {
+        const {connected, ticketDetails, isLoaded: isPropsLoaded} = this.props;
+        const {ticketId, isLoaded: isStateLoaded} = this.state;
+        if (connected && !isStateLoaded && ((ticketDetails && ticketDetails.ticketId) !== ticketId || !isPropsLoaded)) {
+            this.init();
+        } else if (connected && !isStateLoaded && ticketDetails && ticketDetails.ticketId === ticketId) {
+            this.setTicket(this.props);
+        }
+    }
+
+    componentDidUpdate() {
+        const {isLoaded: isPropsLoaded, ticketDetails} = this.props;
+        const {ticketId, isLoaded: isStateLoaded} = this.state;
+        if (isPropsLoaded && !isStateLoaded && ticketDetails && ticketDetails.ticketId === ticketId) {
+            this.setTicket(this.props);
+        }
+    }
+
+    setTicket(data: Props) {
+        this.setState({
+            isLoaded: true,
+            ticketDetails: data.ticketDetails,
+        });
+    }
+
+    fixVersionLabel(fixVersion: string) {
+        if (fixVersion) {
+            const fixVersionString = 'Fix Version :';
+            return (
+                <div className='fix-version-label'>
+                    {fixVersionString}
+                    <span className='fix-version-label-value'>
+                        {fixVersion}
+                    </span>
+                </div>
+            );
+        }
+
+        return (<span/>);
+    }
+
+    tagTicketStatus(ticketStatus: string) {
+        let ticketStatusClass = 'default-style';
+
+        const myStatusClass = myStatusClasses[ticketStatus.toLowerCase()];
+        if (myStatusClass) {
+            ticketStatusClass += myStatusClass;
+        } else {
+            ticketStatusClass += ' ticket-status--default';
+        }
+
+        return <span className={ticketStatusClass}>{ticketStatus}</span>;
+    }
+
+    labelList(labels: string[]) {
+        if (!labels.length) {
+            return null;
+        }
+
+        let totalString = 0;
+        let totalHide = 0;
+        const labelList = labels.map((label: string, key: number): JSX.Element => {
```

Comment:
This way you can return `null` below instead of `<></>`

```suggestion
        const labelList = labels.map((label: string, key: number): React.ReactNode => {
```